### PR TITLE
Reset password modal

### PR DESCRIPTION
**Pull request by sterlingwellscaffeine**
Adds the forgot password prompt to the login screen
Adds a reusable modal which in this case is used to reset passwords

Aligns with the mocks that can be found here: https://docs.google.com/document/d/1bmTLGs8nROB2iOC9-87ZU_IgkPldnPiKUWz6crQxJbw/edit?usp=sharing

### CODE REVIEW

File: src/interface/src/app/login/login.component.html [hunk 1]
```diff
@@ -14,6 +14,17 @@ <h1 class="signin-title">Sign in to Planscape</h1>
           <input type="password" required formControlName="password" matInput />
         </mat-form-field>
 
+        <button
+          mat-flat-button
+          (click)="openDialog()"
+          aria-label="Opens a dialog for users who have forgotten their password"
+          i18n-aria-label="
```
Comment (sachiyuan) [inline src/interface/src/app/login/login.component.html:21]: Looks like this is causing issues with some of the tests. Here's the output from one if you don't know how to access it yet:
Chrome Headless 116.0.5845.96 (Linux x86_64) LoginComponent continue as guest navigates to home page FAILED
	Error: It looks like your application or one of its dependencies is using i18n.
	Angular 9 introduced a global `$localize()` function that needs to be loaded.
	Please run `ng add @angular/localize` from the Angular CLI.
	(For non-CLI projects, add `import '@angular/localize/init';` to your `polyfills.ts` file.
	For server-side rendering applications add the import to your `main.server.ts` file.)
	    at _global.$localize (node_modules/@angular/core/fesm2020/core.mjs:29729:15)
	    at constsOrFactory (ng:///LoginComponent.js:35:18)
	    at createTView (node_modules/@angular/core/fesm2020/core.mjs:12175:60)
	    at getOrCreateComponentTView (node_modules/@angular/core/fesm2020/core.mjs:12150:28)
	    at createRootComponentView (node_modules/@angular/core/fesm2020/core.mjs:13976:49)
	    at ComponentFactory.create (node_modules/@angular/core/fesm2020/core.mjs:13835:35)
	    at apply (node_modules/@angular/core/fesm2020/testing.mjs:26224:51)
	    at _ZoneDelegate.invoke (node_modules/zone.js/fesm2015/zone.js:372:26)
	    at ProxyZoneSpec.onInvoke (node_modules/zone.js/fesm2015/zone-testing.js:287:39)
	    at _ZoneDelegate.invoke (node_modules/zone.js/fesm2015/zone.js:371:52)
	    
	    You can run these tests locally by running `ng test`

Comment (sterlingwellscaffeine) [reply src/interface/src/app/login/login.component.html:21]: I have fixed the tests that I broke, note that the existing tests are not in a stable state on main.

**Comment by bwndai** [comment]
Have you considered dj_rest_auth.views.PasswordResetView? documentation: https://dj-rest-auth.readthedocs.io/en/latest/configuration.html#password-reset-serializer, example: https://stackoverflow.com/questions/53945056/django-rest-auth-password-reset

File: src/interface/src/app/login/login.component.html [hunk 2]
```diff
@@ -14,16 +14,25 @@ <h1 class="signin-title">Sign in to Planscape</h1>
           <input type="password" required formControlName="password" matInput />
         </mat-form-field>
 
+        <mat-card-actions>
+          <a
```
Comment (macieksmuga) [inline src/interface/src/app/login/login.component.html:18]: Why not a button? I seem to recall that a11y validators will reject <a> tags with (click) actions?

File: src/interface/src/app/login/login.component.spec.ts [hunk 3]
```diff
@@ -20,12 +22,13 @@ describe('LoginComponent', () => {
...
     );
     TestBed.configureTestingModule({
-      imports: [FormsModule, ReactiveFormsModule],
+      imports: [FormsModule, ReactiveFormsModule, MatDialogModule,],
       declarations: [LoginComponent],
       schemas: [CUSTOM_ELEMENTS_SCHEMA],
       providers: [
         { provide: Router, useFactory: routerStub },
         { provide: AuthService, useValue: fakeAuthService },
+
```
Comment (macieksmuga) [inline src/interface/src/app/login/login.component.spec.ts:31]: Add test to validate dialog popping up?

**Comment by sterlingwellscaffeine** [comment]
> 

We discussed this offline but essentially these are links to the top couple google search entries for Django password auth and after deeper investigation are primarily focused on the server code and have no relevancy to the front end screen implementation. Marking as won't fix.

File: src/interface/src/app/login/login.component.ts [hunk 4]
```diff
@@ -48,6 +51,14 @@ export class LoginComponent {
     });
   }
 
+  resetPassword() {
+
+    // TODO: sterlingwellscaffeine -- add the functionality that validates the 
+    // email. The reason that this is not included in this CL is that I would like
+    // to make that a part of the server call and the server is not yet up and running.
+    this.dialog.open(ResetPasswordDialog);
```
Comment (macieksmuga) [inline src/interface/src/app/login/login.component.ts:59]: Do we want to leave users on this page or redirect them to the homepage or somesuch (as they'll come back to the app via their email link, so it doesn't make sense to me to leave this login page open)

**Review by macieksmuga — CHANGES_REQUESTED** [review]

File: src/interface/src/app/login/login.component.html (hunk 2 above)
Comment (sterlingwellscaffeine) [reply src/interface/src/app/login/login.component.html:18]: The mocks that I was given showed that it was an anchor element, so I am following along with the mocks.

File: src/interface/src/app/login/login.component.ts (hunk 4 above)
Comment (sterlingwellscaffeine) [reply src/interface/src/app/login/login.component.ts:59]: Did the demo help explain the functionality to you? The user is never leaving the homepage.

File: src/interface/src/app/login/login.component.spec.ts (hunk 3 above)
Comment (sterlingwellscaffeine) [reply src/interface/src/app/login/login.component.spec.ts]: Done

File: src/interface/src/app/login/login.component.ts (hunk 4 above)
Comment (macieksmuga) [reply src/interface/src/app/login/login.component.ts:59]: Could you paste demo screenshots in the PR description?

**Review by macieksmuga — APPROVED** [review]

**Pull request merged**